api/src/test/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAOTest.java

```diff
@@ -0,0 +1,134 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Assert;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.rules.ExpectedException;
+import org.openmrs.*;
+import org.openmrs.api.db.DiagnosisDAO;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateDiagnosisDAOTest extends BaseContextSensitiveTest {
+
+	private static final String DIAGNOSIS_XML = "org/openmrs/api/include/HibernateDiagnosisDAOTestDataset.xml";
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Autowired
+	DiagnosisDAO diagnosisDAO;
+
+	@Before
+	public void setUp() {
+		executeDataSet(DIAGNOSIS_XML);
+
+	}
+
+	@Test
+	public void shouldSaveDiagnosis() {
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText(new Concept(4),
+			new ConceptName(5089), "non coded");
+		Date date = new Date();
+		int id = 5;
+		String uuid = "4e663d96-6b78-11e0-93c3-18a9b5e044dc";
+		int rank = 2;
+		ConditionVerificationStatus certainty = ConditionVerificationStatus.CONFIRMED;
+		User creator = new User(1);
+		Condition condition = new Condition();
+		Patient patient = new Patient(2);
+		Encounter encounter = new Encounter(3);
+		
+		Diagnosis diagnosis = new Diagnosis();
+		diagnosis.setEncounter(encounter);
+		diagnosis.setRank(rank);
+		diagnosis.setId(id);
+		diagnosis.setCertainty(certainty);
+		diagnosis.setPatient(patient);
+		diagnosis.setCondition(condition);
+		diagnosis.setDiagnosis(codedOrFreeText);
+		diagnosis.setUuid(uuid);
+		diagnosis.setCreator(creator);
+		diagnosis.setVoided(false);
+		diagnosis.setDateCreated(date);
+
+		diagnosisDAO.saveDiagnosis(diagnosis);
+
+		Diagnosis savedDiagnosis = diagnosisDAO.getDiagnosisById(id);
+		
+		assertEquals(uuid, diagnosis.getUuid());
+		assertEquals(false, diagnosis.getVoided());
+		assertEquals(rank, (int) diagnosis.getRank());
+		assertEquals(certainty, diagnosis.getCertainty());
+		assertEquals(creator, diagnosis.getCreator());
+		assertEquals(condition, diagnosis.getCondition());
+		assertEquals(patient, diagnosis.getPatient());
+		assertEquals(encounter, diagnosis.getEncounter());
+		
+	}
+
+	@Test
+	public void shouldDiagnosisByUuid() {
+		String uuid = "4e663d66-6b78-11e0-93c3-18a905e044dc";
+		int diagnosisId = 1;
+
+		Diagnosis diagnosis = diagnosisDAO.getDiagnosisByUuid(uuid);
+		assertEquals(diagnosisId, (int) diagnosis.getId());
+	}
+
+	@Test
+	public void shouldDiagnosisById() {
+		int diagnosisId = 1;
+		String uuid = "4e663d66-6b78-11e0-93c3-18a905e044dc";
```

Comment:
You do not need to declare variables if you are going to use them in just one place